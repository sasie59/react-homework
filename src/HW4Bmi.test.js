import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import HW from './HW4Bmi';
test('should HW4Bmi', () => {
  const {container } =render(<HW />);
  expect(container).toMatchSnapshot;
});

test('should 體重/身高平方(公尺)', () => {
  const {container } =render(<HW />);
  expect(60/Math.pow(1.7)).toBe(Math.round(20.76));
  expect(container).toMatchSnapshot();
});
test('should 體重/身高平方(公尺)', () => {
  const {container } =render(<HW />);
  expect(80/Math.pow(1.9)).toBe(Math.round(20.76));
  expect(container).toMatchSnapshot(Math.round(20.16));
  expect(container).toMatchSnapshot();
});
