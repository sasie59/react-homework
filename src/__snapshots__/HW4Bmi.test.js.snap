// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should 90kg / 170cm get 31 1`] = `
<div>
  <div>
    <h1>
      複習React & Hooks & Jest
    </h1>
    <form
      action=""
    >
      體 重
      <input
        name="kg"
        type="text"
      />
      <br />
      身 高
      <input
        name="cm"
        type="text"
      />
    </form>
    <button>
      計算!
    </button>
    <h1>
      BMI : 
      <span
        data-testid="bmi"
      />
    </h1>
  </div>
</div>
`;

exports[`should HW4Bmi 1`] = `
<div>
  <div>
    <h1>
      HW4Bmi
    </h1>
    <form
      action=""
    >
      體重
      <input
        name="kg"
        type="text"
      />
      <br />
      身高
      <input
        name="cm"
        type="text"
      />
    </form>
    <button>
      點擊
    </button>
    <h2>
      BMI :
      <span />
    </h2>
  </div>
</div>
`;
